packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -54,6 +58,82 @@ List<String> debugDescribeTransform(Matrix4 transform) {
   return matrix;
 }
 
+bool _extensionsInitialized = false;
+
+void initServiceExtensions() {
+  if (_extensionsInitialized)
+    return;
+
+  _extensionsInitialized = true;
+
+  assert(() {
+    registerExtension('flutter', _flutter);
+    // TODO: Expose debugDumpApp().
+    // TODO: Expose StatisticsOverlay.
+    registerExtension('flutter.debugPaint', _debugPaint);
+    registerExtension('flutter.timeDilation', _timeDilation);
+
+    // Emit an info level log message; this tells the debugger that the Flutter
+    // service extensions are registered.
+    log('Flutter initialized', name: 'flutter', level: 800);
+
+    return true;
+  });
+}
+
+/// Just respond to the request. Clients can use the existence of this call to
+/// know that the debug client is a Flutter app.
```

Comment:
Yup, that's the case. I'm thinking of a system where the bare `flutter` call would return the the client some metadata about the system - possibly a version - and a list of all the flutter commands that are available. That way the client would know which commands were available w/o having to make speculative calls to each command.

If the service protocol had a command to query what service extensions were registered, that would work as well. The service protocol doesn't have this functionality, but we could ask about the feasibility of adding it.